dpnp/dpnp_iface_statistics.py

```diff
@@ -352,69 +350,101 @@ def histogram(a, bins=10, range=None, density=None, weights=None):
     >>> a = np.arange(4).reshape((2,2))
-    >>> a.shape
-    (2, 2)
-    >>> [i for i in a]
-    [0, 1, 2, 3]
+    >>> a
+    array([[0, 1],
+           [2, 3]])
     >>> np.max(a)
-    3
+    array(3)
+
+    >>> np.max(a, axis=0)   # Maxima along the first axis
+    array([2, 3])
+    >>> np.max(a, axis=1)   # Maxima along the second axis
+    array([1, 3])
+
+    >>> b = np.arange(5, dtype=float)
+    >>> b[2] = np.NaN
+    >>> np.max(b)
+    array(nan)
 
     """
 
-    x1_desc = dpnp.get_dpnp_descriptor(x1, copy_when_nondefault_queue=False)
-    if x1_desc:
-        # Negative values in 'shape' are not allowed in input array
-        # 306-322 check on negative and duplicate axis
-        isaxis = True
-        if axis is not None:
-            if dpnp.isscalar(axis):
-                if axis < 0:
-                    isaxis = False
-            else:
-                for val in axis:
-                    if val < 0:
-                        isaxis = False
-                        break
-                if isaxis:
-                    for i in range(len(axis)):
-                        for j in range(len(axis)):
-                            if i != j:
-                                if axis[i] == axis[j]:
-                                    isaxis = False
-                                    break
-
-        if not isaxis:
-            pass
-        elif out is not None:
-            pass
-        elif keepdims:
-            pass
-        elif initial is not None:
-            pass
-        elif where is not True:
-            pass
+    if initial is not None:
+        raise NotImplementedError(
+            "initial keyword argument is only supported by its default value."
+        )
+    elif where is not True:
+        raise NotImplementedError(
+            "where keyword argument is only supported by its default values."
+        )
+    else:
+        dpt_array = dpnp.get_usm_ndarray(a)
+        if dpt_array.size == 0:
+            # TODO: get rid of this if condition when dpctl supports it
+            for i in range(a.ndim):
+                if a.shape[i] == 0:
+                    if axis is None or i in axis:
+                        raise ValueError(
+                            "reduction does not support zero-size arrays"
+                        )
+                    else:
+                        indices = [i for i in range(a.ndim) if i not in axis]
+                        res_shape = tuple([a.shape[i] for i in indices])
+                        result = dpnp.empty(res_shape, dtype=a.dtype)
         else:
-            result_obj = dpnp_max(x1_desc, axis).get_pyobj()
-            result = dpnp.convert_single_elem_array_to_scalar(result_obj)
-
+            result = dpnp_array._create_from_usm_ndarray(
+                dpt.max(dpt_array, axis=axis, keepdims=keepdims)
+            )
+        if out is None:
             return result
+        else:
+            if out.shape != result.shape:
+                raise ValueError(
```

Comment:
please increase the coverage by tests with negative use cases (the same for `min`)